azure/petctl.py

```diff
@@ -0,0 +1,348 @@
+from util import *
+import argparse
+
+# Create a Kubernetes specs and YAML job file based on user inputs
+def configure(args):
+    configure_yaml(args)
+    configure_json(args)
+
+
+# Deploys a Kubernetes cluster
+def setup(args):
+    # Login to Azure
+    azure_login()
+    # Install AKS Engine
+    install_aks_engine()
```

Comment:
Curious why you need AKS Engine instead of vanilla AKS?

Reply:
We used AKS Engine instead of the vanilla AKS because in AKS, master node is a managed service which user doesn’t have any control over. This means that user has to configure one of the workers as a master and this complicates deployment of heterogenous nodes. With AKS Engine it was easier for us to provision a heterogenous multi-node cluster.

Reply:
Just to confirm (and sorry if I missed this detail) - are you using the k8s master for etcd rendezvous for torch elastic? Otherwise, there shouldn't be a need to customize the master in most cases.

Reply:
Yes, aks-engine installs etcd in the master node of the Kubernetes cluster.

Reply:
Thank you! Just a couple of thoughts to consider: AKS Engine is a bit of an advanced tool and many AKS users might not be aware of it and its trade-offs. For what it's worth, many operators that need to talk to the control plane are supported under AKS. My suggestion is to either talk to the AKS master etcd or install in-cluster etcd as is done for the AWS implementation.

Reply:
Agreed. We used aks-engine because master node isn't exposed to workers in AKS. We are planning to install in-cluster etcd for our AzureML implementation.